data_validation/src/spark_rapids_validation_tool/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,273 @@
+    |       ...
+    |       coln|1.23456789E8|9.87654321E8|5.94837261E8|4.513124419597775E8|                2.0|
+    +-----------+------------+------------+------------+-------------------+-------------------+
+    """
+    result = None
+    agg_functions = [min, max, avg, stddev, countDistinct]
+    # if not specified any included_columns, then get all numeric cols
+    excluded_columns_list = [e.strip() for e in e.split(",")]
+    metrics_cols = [i.strip() for i in i.split(",") if i not in excluded_columns_list]
+    if i in ['None', 'all']:
+        metrics_cols = [c.name for c in table_DF.schema.fields if
+                        any(fnmatch.fnmatch(c.dataType.simpleString(), pattern) for pattern in ['*int*', '*decimal*', '*float*', '*double*'])]
+    for col in metrics_cols:
+        dfc = spark.createDataFrame(([col],), ["ColumnName"])
+        table1_agg = table_DF.select(
+            [f(col).alias(f.__name__ + t1) for f in
+             agg_functions])
+        tmp_df = dfc.join(table1_agg)
+        if result is None:
+            result = tmp_df
+        else:
+            result = result.union(tmp_df)
+    return result
+
+def metrics_metadata(spark, format, t1, t2, t1p, t2p, pk, i, e, f, p):
+    """
+    The different metadata of each column in each table(min/max/avg/stddev/count_distinct):
+    (If the values are identical, then a specific cell is empty, aka NULL. So we only show differences),
+    the result dataframe should be:
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    |ColumnName| min_A| min_B| max_A| max_B|            avg_A|            avg_B|           stddev_A|           stddev_B|countdist_A|countdist_B|
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    |      col1|      |      |12.000|11.000|6.090909090909091|              6.0|  3.477198454346414|    3.3166247903554|           |           |
+    |      col3|      |      |      |      |5.090909090909091|5.181818181818182| 3.1766191290283903|  3.060005941764879|           |           |
+    |      col8|12.340|12.330|      |      |       12.4345455|       12.4336364|0.30031195901474267|0.30064173786328213|          3|          4|
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    """
+    table1_DF = load_table(spark, format, t1, t1p, pk, e, i, f, "")
+    table2_DF = load_table(spark, format, t2, t2p, pk, e, i, f, "")
+
+    table_metric_df1 = generate_metric_df(spark, table1_DF, i, e, t1)
+    table_metric_df2 = generate_metric_df(spark, table2_DF, i, e, t2)
+    joined_table = table_metric_df1.alias("t1").join(table_metric_df2.alias("t2"), ["ColumnName"])
+
+    cond = (round("t1.min" + t1, p) != round("t2.min" + t2, p)) | \
```

Comment:
Wondering how difficult would it be to extend to other data types(string, map) with this approach? Should we have a different `condition` for string and then append that to result_table?

Reply:
will draft a follow up pr to extend to other data types(map/struct), this still depends on customers requirements.

Reply:
add supported for str and map/nested